plugin/federation/federation.gotpl

```diff
@@ -78,41 +84,66 @@ func (ec *executionContext) __resolve_entities(ctx context.Context, representati
 		}()
 
 		switch typeName {
-		{{ range .Entities }}
-			{{ if .ResolverName }}
+		{{ range $_, $entity := .Entities }}
+			{{- if .Resolvers -}}
 				{{ if not .Multi -}}
-
-			case "{{.Def.Name}}":
-				{{ range $i, $keyField := .KeyFields -}}
-					id{{$i}}, err := ec.{{.Type.UnmarshalFunc}}(ctx, rep["{{.Field.Join `"].(map[string]interface{})["`}}"])
-					if err != nil {
-						return errors.New(fmt.Sprintf("Field %s undefined in schema.", "{{.Definition.Name}}"))
-					}
-				{{end}}
-
-				entity, err := ec.resolvers.Entity().{{.ResolverName | go}}(ctx,
-					{{ range $i, $_ := .KeyFields -}} id{{$i}}, {{end}})
-				if err != nil {
-					return err
-				}
-
-				{{ range .Requires }}
-					entity.{{.Field.JoinGo `.`}}, err = ec.{{.Type.UnmarshalFunc}}(ctx, rep["{{.Field.Join `"].(map[string]interface{})["`}}"])
+				case "{{.Def.Name}}":
+					resolverName, err := ec.entityResolverNameFor{{.Def.Name}}(ctx, rep)
 					if err != nil {
-						return err
+						return fmt.Errorf(`finding resolver for Entity "{{.Def.Name}}": %w`, err)
 					}
-				{{ end }}
-
-				list[idx[i]] = entity
-				return nil
+					{{ if eq (len .Resolvers) 1 }}
```

Comment:
I think it should be OK to always generate the switch statement. There isn't a huge runtime performance penalty and it keep this already giant template more manageable and less error prone since you have to make any template changes in multiple places.

Or is there a reason you'd prefer to having a different generated code when there is only 1 entity vs multiple?

Reply:
Only that it made the generated code cleaner. But you're right, it comes at the cost of a more complicated template, which will probably need to be comprehended more often. I'm removing the ==1 case.

Reply:
There were two places where this could be done, hopefully strikes a good balance.